### PR TITLE
147.refactor vibes and social situations to use enums

### LINKED ISSUE #147: Refactor Vibes and Social Situations to Use Enums
[//]: # (These are comments and are used for reference and will not show up in the PR)

User Requirement: N/A

[//]: # (Please Add labels for type of issue)

Description: Decided that it is more maintainable to refactor all our vibes and social situations to completely use enums. Vibes are currently implemented using a factory pattern, but it may result in some difficulties with maintaining the code associated with said code. 

NOTE: This should be done after all user requirement issues/stories have been completed as part of refactoring.

[//]: # (If an issue is a bug or defect, please list steps to be able to reproduce bug and a screenshot of the defect i.e terminal message)

Story Points: 5

Related Issues: N/A

[//]: # (Notes:)
[//]: # (- Testing and Documentation tasks should always be assumed, but only need to be explicitly stated if there is something that ~~does not~~ need modification. For example,)
[//]: # (some Issues/Stories may not require one of Unit or Intent tests, and this should be indicated in the task list.)
[//]: # (- Code Reviews are also always assumed to be required, but an issue should only indicate in the description if a review is not required, but someone still shouldn't)
[//]: # (merge their own pull request)

### PR DESCRIPTION
**Pull request by PranavB6**
[//]: # (These are comments and are used for reference and will not show up in the PR)

Description: 
* I've integrated enums for vibes and social situation. 

* For vibes, we no longer need the vibe factory or all the difference classes of vibes - it's all now in one file call Vibe! 
  * NOTE: I chose to implement it in a way such that if the database contains a vibe that is not recognized by the app, the app considers it a "blank" vibe. We can change this if you guys want ofc.

* For Social Situation, the code is a little more cleaner.
    * NOTE: I chose to implement it in a way such that if the database contains a social situation that is not recognized, the app will ignore it and just not show it. User's can still edit the social situation and override it. 

* Tests for vibe event and vibe factory have been refactored/rewritten to reflect changes

MOST IMPORTANTLY: I changed the Enum values for Social Situation to match what the convention of Enums is. This means that the social situations in the vibe events we've written so far won't be recognized. (The app won't crash or anything, it will just ignore it and treat it like it doesn't have a social situation).

[//]: # (automatically closes given issue number)
Closes #147 

Checklist

[//]: # (Add an x between square brackets to mark item as done)

- [x] Follows style guidelines and is **properly commented**
- [x] Code changes generate no new warnings
- [x] Testing (Unit and/or Intent)
- [x] ~Documentation (UML, Wiki, etc.)~

### CODE REVIEW

File: app/src/main/java/com/example/vybe/MapFragment.java [hunk 1]
```diff
@@ -24,8 +24,6 @@
 import androidx.fragment.app.Fragment;
 
 import com.example.vybe.Models.VibeEvent;
-import com.example.vybe.Models.vibefactory.Vibe;
```
Comment (jacobwong98) [inline app/src/main/java/com/example/vybe/MapFragment.java]: oof

File: app/src/main/java/com/example/vybe/Models/SocialSituation.java [hunk 2]
```diff
@@ -1,58 +1,46 @@
...
 import androidx.annotation.NonNull;
 
 import java.util.ArrayList;
 
 public enum SocialSituation {
-    Alone("Alone"),
-    WithAnotherPerson("With one other person"),
-    WithSeveralPeople("With two or several people"),
-    WithCrowd("With a crowd"),
-    Same("Same");
```
Comment (jacobwong98) [inline app/src/main/java/com/example/vybe/Models/SocialSituation.java]: same

File: app/src/main/java/com/example/vybe/Models/SocialSituation.java [hunk 3]
```diff
@@ -1,58 +1,46 @@
...
         return SocialSituation.values()[position];
     }
 
-    /**
-     * Get a SocialSituation using its desc
-     * @param desc
-     * Description of the SocialSituation Enum
-     * @return SocialSituation that has the description desc
-     */
-    public static SocialSituation of(String desc) {
```
Comment (jacobwong98) [inline app/src/main/java/com/example/vybe/Models/SocialSituation.java]: why it named "of"?

File: app/src/main/java/com/example/vybe/Models/SocialSituation.java [hunk 4]
```diff
@@ -69,4 +57,27 @@ public static SocialSituation of(String desc) {
...
+    /**
+     * Get the Social Situation of a string. If no matching Social Situation is found, return null.
+     * @param name
+     * Name of Social Situation eg. ALONE, WITH_ANOTHER_PERSON, etc..
+     * NOTE: "Alone", "With another person", etc... are not the names of Social Situation and this function will return NULL if they are entered
+     * @return
+     * A SocialSocituation using it's name
+     */
+
+    public static SocialSituation of(String name) {
```
Comment (jacobwong98) [inline app/src/main/java/com/example/vybe/Models/SocialSituation.java]: why is this just named "of"

File: app/src/main/java/com/example/vybe/AddEdit/VibeCarouselDialogFragment.java [hunk 5]
```diff
@@ -95,10 +96,7 @@ public void onPageScrollStateChanged(int state) {
                 .setTitle("Select a Vibe")
                 .setNegativeButton("Cancel", null)
                 .setPositiveButton("OK", (dialog, which) -> {
-                    // if no vibe selected just exit
-                    if (selectedEmoticon != R.drawable.ic_no_vibe) {
-                        onVibeSelectedListener.onVibeSelected(VibeFactory.getVibe(selectedEmoticon));
-                    }
+                    onVibeSelectedListener.onVibeSelected(Vibe.ofEmoticon(selectedEmoticon));
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/AddEdit/VibeCarouselDialogFragment.java]: gross

Comment (raysarinas) [reply app/src/main/java/com/example/vybe/AddEdit/VibeCarouselDialogFragment.java]: @TheBroMoe does not approve either

File: app/src/test/java/com/example/vybe/VibeFactoryTest.java [hunk 6]
```diff
@@ -1,7 +1,6 @@
 package com.example.vybe;
```
Comment (raysarinas) [inline app/src/test/java/com/example/vybe/VibeFactoryTest.java]: Rename this to VibeEnumTest or something whatever

File: app/src/test/java/com/example/vybe/VibeFactoryTest.java [hunk 7]
```diff
@@ -12,78 +11,120 @@
...
-        assertNull(vibeless);
     }
 
-    @Test
-    public void GetNullVibe_NonEmpty() {
-        Vibe antivibe = VibeFactory.getVibe("Antivibe");
-
-        assertNull(antivibe);
+    @Test(expected = IllegalArgumentException.class)
+    public void GetInValidVibe_ThrowException() {
```
Comment (raysarinas) [inline app/src/test/java/com/example/vybe/VibeFactoryTest.java]: From @TheBroMoe "InValid" should not have the V capitalized

File: app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java [hunk 8]
```diff
@@ -17,7 +17,7 @@
 
 import com.bumptech.glide.Glide;
 import com.example.vybe.MapFragment;
-import com.example.vybe.Models.SocialSituation;
+import com.example.vybe.Models.SocSit;
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java]: wait why is this renamed to `SocSit` I thought we were just wanting to change all the variable stuff? Unless this has to do with how everything is stored in the database?

File: app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java [hunk 9]
```diff
@@ -39,7 +39,7 @@
  * This Activity displays the screen for a user to add a vibe event, or
  * edit an existing vibe event by adding or modifying the different vibe attributes
  */
-public class AddEditVibeEventActivity extends AppCompatActivity implements SocialSituationFieldFragment.OnSocStnSelectedListener, ImageFieldFragment.OnImageSelectedListener, VibeCarouselDialogFragment.OnVibeSelectedListener, LocationSelectionDialog.OnLocationSelectedListener, MapFragment.OnMapFragmentReadyListener {
+public class AddEditVibeEventActivity extends AppCompatActivity implements SocSitFieldFragment.OnSocSitSelectedListener, ImageFieldFragment.OnImageSelectedListener, VibeCarouselDialogFragment.OnVibeSelectedListener, LocationSelectionDialog.OnLocationSelectedListener, MapFragment.OnMapFragmentReadyListener {
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java]: this makes me uncomfortable why is there so many interfaces being implemented???

File: app/src/main/java/com/example/vybe/Models/Vibe.java [hunk 10]
```diff
@@ -32,7 +33,13 @@
      * @return A Vibe Enum
      */
     public static Vibe ofName(String name) {
-        return Vibe.valueOf(name.toUpperCase());
+        try {
+            return Vibe.valueOf(name.toUpperCase());
+
+        } catch (IllegalArgumentException e) {
+            return Vibe.BLANK;
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/Models/Vibe.java]: is it possible when we return a BLANK vibe that a toast also pops up about having an invalid vibe?

File: app/src/test/java/com/example/vybe/VibeTest.java [hunk 11]
```diff
@@ -1,135 +1,159 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Tests the VibeFactory Class and Subclasses
+ *
+ * These tests ensure that the data of a Vibe is properly stored and returned from a Vibe Object
+ */
+public class VibeTest {
+
```
Comment (raysarinas) [inline app/src/test/java/com/example/vybe/VibeTest.java]: does @TheBroMoe still need to update these tests? Or did you do it already @PranavB6

File: app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java (hunk 8 above)
Comment (PranavB6) [reply app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java]: I just changed everything for consistency

File: app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java (hunk 9 above)
Comment (PranavB6) [reply app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java]: Haha I get what you mean but its for communicating with fragments like we did in our assignment. For example when you select a socsit in the SocSitFieldFragment, the `onSocSitSelected` method of `AddEditVibeActivity` is called.

File: app/src/main/java/com/example/vybe/Models/Vibe.java (hunk 10 above)
Comment (PranavB6) [reply app/src/main/java/com/example/vybe/Models/Vibe.java]: Sure, that's a good idea

Comment (PranavB6) [reply app/src/main/java/com/example/vybe/Models/Vibe.java]: Actually, to create a toast you need context and Vibe being a model class, doesn't have context

File: app/src/test/java/com/example/vybe/VibeTest.java (hunk 11 above)
Comment (PranavB6) [reply app/src/test/java/com/example/vybe/VibeTest.java]: I already did it

File: app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java (hunk 8 above)
Comment (raysarinas) [reply app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java]: I mean I guess that makes sense

File: app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java (hunk 9 above)
Comment (raysarinas) [reply app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java]: Sure. But I think we should talk to Ali if he has a suggestion as to what we could do.

**Pull request merged**